src/Components/Steps/Expenses/Expenses.tsx

```diff
@@ -0,0 +1,323 @@
+  const navigate = useNavigate();
+  const intl = useIntl();
+  const translatedAriaLabel = intl.formatMessage({
+    id: 'questions.hasExpenses-ariaLabel',
+    defaultMessage: 'has expenses',
+  });
+  const backNavigationFunction = () => navigate(`/${uuid}/step-${currentStepId - 1}/${formData.householdSize}`);
+  const nextStep = useGoToNextStep('hasExpenses');
+  const expenseOptions = useConfig('expense_options');
+
+  const oneOrMoreDigitsButNotAllZero = /^(?!0+$)\d+$/;
+  const expenseSourceSchema = z.object({
+    expenseSourceName: z.string().min(1),
+    expenseAmount: z.string().regex(oneOrMoreDigitsButNotAllZero),
+  });
+  const expenseSourcesSchema = z.array(expenseSourceSchema);
+  const hasExpensesSchema = z.string().regex(/^true|false$/);
+  const formSchema = z.object({
+    hasExpenses: hasExpensesSchema,
+    expenses: expenseSourcesSchema,
+  });
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    watch,
+    getValues,
+  } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      hasExpenses: formData.hasExpenses ? 'true' : 'false',
+      expenses: formData.expenses ?? [],
+    },
+  });
+  const watchHasExpenses = watch('hasExpenses');
+  // @ts-ignore
+  const hasTruthyExpenses = watchHasExpenses === 'true' || watchHasExpenses === true;
+  const { fields, append, remove, replace } = useFieldArray({
+    control,
+    name: 'expenses',
+  });
+
+  useEffect(() => {
+    const noExpensesAreListed = Number(getValues('expenses').length) === 0;
+    if (hasTruthyExpenses && noExpensesAreListed) {
+      append({
+        expenseSourceName: '',
+        expenseAmount: '',
+      });
+    }
+
+    if (!hasTruthyExpenses) {
+      replace([]);
+    }
+  }, [getValues('hasExpenses')]);
+
+  const formSubmitHandler: SubmitHandler<z.infer<typeof formSchema>> = async (expensesObject) => {
+    const hasExpensesBoolean: boolean = expensesObject.hasExpenses === 'true';
+    if (uuid) {
+      const updatedFormData = { ...formData, ...expensesObject, hasExpenses: hasExpensesBoolean };
+      setFormData(updatedFormData);
+      await updateScreen(uuid, updatedFormData, locale);
+      nextStep();
+    }
+  };
+
+  const createExpenseMenuItems = (expenseOptions: Record<string, FormattedMessageType>) => {
+    const disabledSelectMenuItem = (
+      <MenuItem value="select" key="disabled-select-value" disabled>
+        <FormattedMessage id="expenseBlock.createExpenseMenuItems-disabledSelectMenuItemText" defaultMessage="Select" />
+      </MenuItem>
+    );
+
+    const menuItemKeys = Object.keys(expenseOptions);
+    const menuItemLabels = Object.values(expenseOptions);
+
+    const menuItems = menuItemKeys.map((menuItemKey, i) => {
+      return (
+        <MenuItem value={menuItemKey} key={menuItemKey}>
+          {menuItemLabels[i]}
+        </MenuItem>
+      );
+    });
+
+    return [disabledSelectMenuItem, menuItems];
+  };
```

Comment:
I think this is a cleaner way to do this.
```suggestion
  const createExpenseMenuItems = () => {
    const disabledSelectMenuItem = (
      <MenuItem value="select" key="disabled-select-value" disabled>
        <FormattedMessage id="expenseBlock.createExpenseMenuItems-disabledSelectMenuItemText" defaultMessage="Select" />
      </MenuItem>
    );

    const dropdownMenuItems = Object.entries(expenseOptions).map(([value, message]) => {
      return (
        <MenuItem value={value} key={value}>
          {message}
        </MenuItem>
      );
    });

    return [disabledSelectMenuItem, menuItems];
  };
```